File: Coding Exercises/Python/Day05/second_largest.py
def secondLargest(nums):
    if len(nums) < 2:
        raise ValueError("Input list must have at least two elements")

    firstLargest = float('-inf')
    secondLargest = float('-inf')

    for num in nums:
        if num > firstLargest:
            secondLargest = firstLargest
            firstLargest = num
        elif num < firstLargest and num > secondLargest:
            secondLargest = num

    return secondLargest

# def secondLargest(nums):
#     if len(nums) < 2:
#         raise ValueError("Input list must have at least two elements")

#     sorted_nums = sorted(nums, reverse=True)
#     return sorted_nums[1]

File: Coding Exercises/TypeScript/Day05/second_largest.ts
function secondLargest(nums: number[]): number {
  if (nums.length < 2) {
      throw new Error("Input array must have at least two elements");
  }

  let firstLargest: number = Number.NEGATIVE_INFINITY;
  let secondLargest: number = Number.NEGATIVE_INFINITY;

  for (const num of nums) {
      if (num > firstLargest) {
          secondLargest = firstLargest;
          firstLargest = num;
      } else if (num < firstLargest && num > secondLargest) {
          secondLargest = num;
      }
  }

  return secondLargest;
}

// function secondLargest(nums: number[]): number {
//   if (nums.length < 2) {
//       throw new Error("Input array must have at least two elements");
//   }

//   const sortedNums = nums.slice().sort((a, b) => b - a);
//   return sortedNums[1];
// }